modules/ballerina-core/src/main/java/org/ballerinalang/util/semantics/SemanticAnalyzer.java

```diff
@@ -456,6 +456,21 @@ public void visit(BallerinaFunction function) {
 
             BlockStmt blockStmt = function.getCallableUnitBody();
             blockStmt.accept(this);
+
+            if (function.getReturnParameters().length > 0) {
```

Comment:
Can we put this block of code into the method and use it for both functions and actions?